File: KHost.App/Models/QueuedSinger.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace KHost.App.Models
{
    public class QueuedSinger : BaseModel, IModelWithPosition, IModelWithId
    {
        public int? Id { get; set; }

        public float Position { get; set; }

        public int SingerId { get; set; }

        public Singer Singer { get; set; }

        [NotMapped]
        public int QueuedSongsCount { get; set; }
    }
}

File: KHost.App/ClientApp/src/app/models/QueuedSinger.ts
import { ModelWithId } from "./ModelWIthId";
import { ModelWithPosition } from "./ModelWithPosition";
import { Singer } from "./Singer";

export class QueuedSinger implements ModelWithId, ModelWithPosition {

  id?: number;

  singerId?: number;
    
  singer?: Singer;
  
  queuedSongsCount: number = 0;

  position: number = 0;
}